tools/nvqpp/nvq++.in

```diff
@@ -409,6 +420,18 @@ while [ $# -ne 0 ]; do
 	--lambda-lifting)
 		ENABLE_LAMBDA_LIFTING=true
 		;;
+	--opt-plugin)
+		CUDAQ_OPT_PLUGIN_ARGS="${CUDAQ_OPT_PLUGIN_ARGS} --load-cudaq-plugin $2"
+		shift
+		;;
+	--opt-func-pass)
+		CUDAQ_OPT_EXTRA_PASSES=$(add_pass_to_pipeline "${CUDAQ_OPT_EXTRA_PASSES}" "func.func($2)") 
+		shift
+		;;
+	--opt-module-pass)
+		CUDAQ_OPT_EXTRA_PASSES=$(add_pass_to_pipeline "${CUDAQ_OPT_EXTRA_PASSES}" "$2") 
+		shift
```

Comment:
I'm not sure about having two command-line options here. Ideally, we can document that a generic `--opt-pass=`_foo_ command line option requires _foo_ to be in the pass pipeline syntax which is generically `dialect.opname(passname)`. (It's possible to write a pass on the `arith.constant` ops, for example.)

Reply:
I was a bit unsure when doing this as well but I think you're right: we better let the user specify the pass in full, e.g., `func.func(my-custom-pass)`. Will fix it accordingly.

Reply:
ahh.. I see, I will wait for the update. thanks!

Reply:
Fixed in https://github.com/NVIDIA/cuda-quantum/pull/1034/commits/19bec7e234c865a79a5ab3028c7abaa6fbe27267 :)